### PR TITLE
Citus activity views should ignore Postgres parallel queries

### PR DESCRIPTION
**Pull request by onderkalaci**
Skip Postgres' parallel workers while getting Citus backends
    
    Due to Postgres' parallel queries, we sometime have duplicate
    entries in citus_worker_stat_activity with global_pid = 0:
    
    ```
    select cp.query, pp.leader_pid, pp.pid, cp.global_pid FROM citus_worker_stat_activity cp JOIN pg_stat_activity pp USING(pid) where pp.query NOT ilike '%dump%' and pp.query NOT ilike '%prepare%';
    query | leader_pid | pid | global_pid
    ------------------------------------------------------------------------------------------+------------+-------+--------------
    SELECT count(*) AS count FROM public.test_102009 test WHERE (a OPERATOR(pg_catalog.=) 1) |             | 83475 | 110000083474
    SELECT count(*) AS count FROM public.test_102009 test WHERE (a OPERATOR(pg_catalog.=) 1) | 83475       | 83503 | 0
    SELECT count(*) AS count FROM public.test_102009 test WHERE (a OPERATOR(pg_catalog.=) 1) | 83475
    ```
    
    With this commit, we prevent those.

### CODE REVIEW

File: src/backend/distributed/transaction/backend_data.c [hunk 1]
```diff
@@ -401,8 +406,16 @@ StoreAllActiveTransactions(Tuplestorestate *tupleStore, TupleDesc tupleDescripto
 			continue;
 		}
 
+		PGPROC *currentProc = &ProcGlobal->allProcs[backendIndex];
+		if (currentProc->lockGroupLeader != NULL)
```
Comment (onurctirtir) [inline src/backend/distributed/transaction/backend_data.c:410]: or should we use `IsBackgroundWorker` or something similar ?

Comment (onderkalaci) [reply src/backend/distributed/transaction/backend_data.c:410]: The name of the UDF is `citus_worker_stat_activity ` , though we are planning to change it. The idea is to be able show ALL the activity going on the cluster. So, we should be able to show the background workers.

Though, thinking on this again, I think we should also be able to show parallel queries on that view as pg_stat_activity does. We might consider adding `leader_pid` to the output, such that a higher level UDF/function can filter if needed.

In other words, I decided the close this PR for now.
